### ISSUE TITLE
vSphere CSI Chart images end of life.

### ISSUE THREAD

**Issue by phillipsj** [rancher/rancher#37120]
**Is your feature request related to a problem? Please describe.**

Upstream vSphere CSI images have published EOL dates for specific versions. You can find those listed [here](https://docs.vmware.com/en/VMware-vSphere-Container-Storage-Plug-in/2.0/vmware-vsphere-csp-getting-started/GUID-D4AAD99E-9128-40CE-B89C-AD451DA8379D.html?hWord=N4IghgNiBcIMYHsB2AXMBLJBTATgAgGcUEcwBzLPAWzBR3QA8QBfIA). 

Starting in August 2022 any CSI versions older than 2.2.2 will be EOL. Shortly after that in November 2022, 2.3.0 and 2.3.1 will be EOL.

Once all releases older than 2.4.0 have been EOL we will lose upstream support for Kubernetes versions older than 1.20.

While some older versions haven't been deprecated or EOL, they are not versions we currently publish as part of [Rancher charts](https://github.com/rancher/charts/tree/dev-v2.6/charts/rancher-vsphere-csi).

I believe this may impact our current support matrix, especially regarding 2.5.x versions of Rancher that are still supported.

**Describe the solution you'd like**

Guidance on the choices to make regarding these versions and how we should handle making sure these are deprecated and communicated.

**Comment by phillipsj** [comment]
@Jono-SUSE-Rancher @deniseschannon

**Comment by deniseschannon** [comment]
@sirredbeard Can you come up with how you think this should be proposed/updated?

**Comment by snasovich** [comment]
We've bumped the lowest CSI plugin version to 2.5.4 recently for the lowest version of k8s supported by 2.7.x (1.23.x). According to the [linked documentation](https://docs.vmware.com/en/VMware-vSphere-Container-Storage-Plug-in/2.0/vmware-vsphere-csp-getting-started/GUID-D4AAD99E-9128-40CE-B89C-AD451DA8379D.html?hWord=N4IghgNiBcIMYHsB2AXMBLJBTATgAgGcUEcwBzLPAWzBR3QA8QBfIA) there is not EOL date for 2.5.4 yet and even 2.5.2 is Feb 2024 which is a while from now. Closing this issue.
